### ISSUE TITLE
Rule + Data Bridge  '$events/client_connected' worked failure

### ISSUE THREAD

**Issue by iowarecc** [emqx/emqx#8494]
### What happened?

I try the 5.0.3, when I make a rule for my topic, it just worked fine; but I try the topic '$events/client_connected', there was something wrong when I add a sink, the dashboard show the bridge status 'connecting', and I found some error logs.

### What did you expect to happen?

topic '$events/client_connected' could be passed success the same

### How can we reproduce it (as minimally and precisely as possible)?

just make a rule with a bridge,  the rule topic is '$events/client_connected', as emqx news described.

### Anything else we need to know?

emqx_connector_mqtt_msg.erl

```shell
https://github.com/emqx/emqx/blob/master/apps/emqx_connector/src/mqtt/emqx_connector_mqtt_msg.erl
```

```erlang
%% @doc Estimate the size of a message.
%% Count only the topic length + payload size
-spec estimate_size(msg()) -> integer().
estimate_size(#message{topic = Topic, payload = Payload}) ->
    size(Topic) + size(Payload);
estimate_size(#{topic := Topic, payload := Payload}) ->
    size(Topic) + size(Payload).
```

maybe

```erlang
%% @doc Estimate the size of a message.
%% Count only the topic length + payload size
-spec estimate_size(msg()) -> integer().
estimate_size(#message{topic = Topic, payload = Payload}) ->
    size(Topic) + size(Payload);
estimate_size(#{topic := Topic, payload := Payload}) ->
    size(Topic) + size(Payload);

%% added. This two cases should be match
estimate_size(#{<<"topic">> := Topic, <<"payload">> := Payload}) ->
    size(Topic) + size(Payload);
estimate_size(#{}) ->
    128;
```

### EMQX version

<details>

```console
$ ./bin/emqx_ctl broker
sysdescr  : EMQX
version   : 5.0.3
datetime  : 2022-07-15T19:08:07.278321492+08:00
uptime    : 2 hours, 35 minutes, 28 seconds
```

</details>

### OS version

<details>

```console
$uname -a
Darwin mac4cai 20.6.0 Darwin Kernel Version 20.6.0: Tue Apr 19 21:04:45 PDT 2022; root:xnu-7195.141.29~1/RELEASE_X86_64 x86_64 i386 Darwin
```

</details>

### Log files

<details>

```erlang
[
    {emqx_connector_mqtt_msg,estimate_size,
        [#{<<"clientid">> => <<"hello2">>,<<"username">> => <<"hello1">>}],
        [{file,"emqx_connector_mqtt_msg.erl"},{line,146}]
    },
    {replayq,transform,6,
        [{file,"replayq.erl"},{line,298}]
    },
    {replayq,append,2,[{file,"replayq.erl"},{line,189}]},
    {emqx_connector_mqtt_worker,common,4,[
        {file,"emqx_connector_mqtt_worker.erl"},{line,369}
    ]},
    {gen_statem,loop_state_callback,11,[{file,"gen_statem.erl"},{line,1203}]},
    {proc_lib,init_p_do_apply,3,[{file,"proc_lib.erl"},{line,226}]}
].

{function_clause, [
    {emqx_connector_mqtt_msg,estimate_size,
        [#{<<"clientid">> => <<"hello2">>,<<"username">> => <<"hello1">>}],
        [{file,"emqx_connector_mqtt_msg.erl"},{line,146}]
     },
     {replayq,transform,6,[{file,"replayq.erl"},{line,298}]},
     {replayq,append,2,[{file,"replayq.erl"},{line,189}]},
     {emqx_connector_mqtt_worker,common,4,[{file,"emqx_connector_mqtt_worker.erl"},{line,369}]},
     {gen_statem,loop_state_callback,11,[{file,"gen_statem.erl"},{line,1203}]},
     {proc_lib,init_p_do_apply,3,[{file,"proc_lib.erl"},{line,226}]
    }
]}.

{function_clause,[
    {emqx_connector_mqtt_msg,
        estimate_size,
        [#{<<"clientid">> => <<"hello2">>,<<"username">> => <<"hello1">>}],
        [{file,"emqx_connector_mqtt_msg.erl"},{line,146}]
    },
    {replayq,transform,6,[
        {file,"replayq.erl"},{line,298}
    ]},
    {replayq,append,2,[
        {file,"replayq.erl"},{line,189}
    ]},
    {emqx_connector_mqtt_worker,common,4,
        [{file,"emqx_connector_mqtt_worker.erl"},{line,369}]
    },
    {gen_statem,loop_state_callback,11,
        [{file,"gen_statem.erl"},{line,1203}]
    },
    {proc_lib,init_p_do_apply,3,
        [{file,"proc_lib.erl"},{line,226}]
    }
]}.

```

</details>

**Comment by terry-xiaoyu** [comment]
Will be fixed in 5.0.5.

**Comment by JimMoen** [comment]
It may fixed in the PR.

Note that there is no `topic` and `payload` in event messages such as `$events/client_connected` and `$events/client_disconnected`

For event notification, you may configure more than one mqtt bridge pub topic.
like topic `client/${clientid}/connected` for rule from `$events/client_connected`,
and `client/${clientid}/disconnected` for rule form `$events/client_disconnected`
